gamemode/sv_economy.lua

```diff
@@ -125,19 +126,18 @@ hook.Add("PlayerCanPickupWeapon", "Horde_Economy_Pickup", function (ply, wpn)
     if ply:IsNPC() then return true end
     if HORDE.items[wpn:GetClass()] then
         local item = HORDE.items[wpn:GetClass()]
-        if (ply:GetWeight() - item.weight < 0) or (not item.whitelist[ply:GetClass().name]) then
+        if (ply:GetWeight() - item.weight < 0) or (not item.whitelist[ply:GetClass().name] and not table.HasValue(HORDE.classes[ply:GetClass().name].weapons, wpn:GetClass())) then

```

Comment:
I would leave out this check here. Starter weapons should be configured in the item config to be usable for the class.